Cargo.toml

```diff
@@ -53,3 +55,6 @@ tower = "0.4"
 url = "2.2.0"
 wasm-bindgen = { version = "0.2.83", features = ["serde-serialize"] }
 wasm-bindgen-test = "0.3.33"
+
+[patch.crates-io]
+async-lsp = { git = "https://github.com/oxalica/async-lsp", rev = "9fc2db84ddcda291a864f044657f68d4377557f7" }
```

Comment:
What features are we patching from this commit?

Reply:
`ClientSocket::new_closed()` was added after I requested a way to construct a fake ClientSocket so we can test without creating an entire server/client. I'd rather rely on a patch than delete our tests.

Reply:
Makes sense -- I'm fine with that as we do rely on a forked version of chumsky currently